generators/chipyard/src/main/scala/harness/HarnessClocks.scala

```diff
@@ -36,44 +37,26 @@ trait HarnessClockInstantiator {
   def instantiateHarnessClocks(refClock: ClockBundle): Unit
 }
 
-// The DividerOnlyHarnessClockInstantiator uses synthesizable clock divisors
-// to approximate frequency ratios between the requested clocks
-class DividerOnlyHarnessClockInstantiator extends HarnessClockInstantiator {
-  // connect all clock wires specified to a divider only PLL
-  def instantiateHarnessClocks(refClock: ClockBundle): Unit = {
-    val sinks = _clockMap.map({ case (name, (freq, bundle)) =>
-      ClockSinkParameters(take=Some(ClockParameters(freqMHz=freq / (1000 * 1000))), name=Some(name))
-    }).toSeq
-
-    val pllConfig = new SimplePllConfiguration("harnessDividerOnlyClockGenerator", sinks)
-    pllConfig.emitSummaries()
-
-    val dividedClocks = LinkedHashMap[Int, Clock]()
-    def instantiateDivider(div: Int): Clock = {
-      val divider = Module(new ClockDividerN(div))
-      divider.suggestName(s"ClockDivideBy${div}")
-      divider.io.clk_in := refClock.clock
-      dividedClocks(div) = divider.io.clk_out
-      divider.io.clk_out
-    }
-
-    // connect wires to clock source
-    for (sinkParams <- sinks) {
-      // bypass the reference freq. (don't create a divider + reset sync)
-      val (divClock, divReset) = if (sinkParams.take.get.freqMHz != pllConfig.referenceFreqMHz) {
-        val div = pllConfig.sinkDividerMap(sinkParams)
-        val divClock = dividedClocks.getOrElse(div, instantiateDivider(div))
-        (divClock, ResetCatchAndSync(divClock, refClock.reset.asBool))
-      } else {
-        (refClock.clock, refClock.reset)
-      }
-
-      _clockMap(sinkParams.name.get)._2.clock := divClock
-      _clockMap(sinkParams.name.get)._2.reset := divReset
-    }
-  }
+class ClockSourceAtFreqMHz(val freqMHz: Double) extends BlackBox(Map(
+  "PERIOD" -> DoubleParam(freqMHz)
+)) with HasBlackBoxInline {
+  val io = IO(new ClockSourceIO)
+
+  setInline("ClockSourceAtFreqMHz.v",
```

Comment:
Nit: Can you get the module name in Chisel and set the file name and verilog module name to that? I think we should start to enforce that in inlined modules.

Reply:
Sorry, I'm not quite sure I follow. I thought for black boxes the module name would be stuck at what the Chisel defines it as

Reply:
I see that the filename on L45 and the module name at L47 are hardcoded to `ClockSourceAtFreqMHz`. Can you get the Scala class name w/ `this.getClass.getSimpleName` and put that in both the lines.

Reply:
I also forget what the API in Chisel is to get the module name as well.